dart/gui/osg/Viewer.cpp

```diff
@@ -843,6 +843,53 @@ const ::osg::ref_ptr<::osg::Group>& Viewer::getRootGroup() const
   return mRootGroup;
 }
 
+//==============================================================================
+void Viewer::setVerticalFieldOfView(double fov)
+{
+  double fovy;
+  double aspectRatio;
+  double zNear;
+  double zFar;
+
+  auto* camera = getCamera();
+  assert(camera);
+  const bool result = camera->getProjectionMatrixAsPerspective(
+      fovy, aspectRatio, zNear, zFar);
+
+  if (!result)
+  {
+    dtwarn << "[Viewer::getMasterCameraFieldOfView] Attemping to set vertical "
+           << "field of view while the camera isn't perspective view. "
+           << "Ignoring this request.\n";
+  }
+
+  camera->setProjectionMatrixAsPerspective(fov, aspectRatio, zNear, zFar);
+}
+
+//==============================================================================
+double Viewer::getVerticalFieldOfView() const
+{
+  double fovy;
+  double aspectRatio;
+  double zNear;
+  double zFar;
+
+  const auto* camera = getCamera();
+  assert(camera);
```

Comment:
Does this mean we always assume the return value of `getCamera()` should always be non-null? Is it feasible that it will ever be `nullptr` (e.g. if this `Viewer` doesn't have any cameras)? I'm not finding anything in the OSG docs that indicate what we should expect.

Reply:
Yeah, I also couldn't find related docs from OSG. I think it would be safer we just ignore when `getCamera()` returns null. Let me change this code so. Let me know what you think.